addons/freifunk-berlin-bbbdigger/files/postinst.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+#
+# As part of the install, we don't want to smash an already set up config
+# but at the same time update it if necessary.  The unique attributes are:
+#
+# network.bbbdigger_dev.macaddr
+# tunneldigger.bbbdigger.uuid
+#
+# All other config sections are overwritten with current settings
+
+. /lib/functions.sh
+
+TUNNEL_SERV='77.87.51.51:8942'
+IFACE=bbbdigger
+
+# tunneldigger UUID (and MAC) generation, if there isn't one already
+MAC=$(uci -q get network.${IFACE}_dev.macaddr)
+if [ $? -eq 1 ]; then
+  # start with b6 for Berliner 6ackbone
+  MAC="b6"
+  for byte in 2 3 4 5 6; do
+    MAC=$MAC`dd if=/dev/urandom bs=1 count=1 2> /dev/null | hexdump -e '1/1 ":%02x"'`
+  done
+fi
+
+UUID=$(uci -q get tunneldigger.${IFACE}.uuid)
+if [ $? -eq 1 ]; then
+  UUID=$MAC
+  for byte in 7 8 9 10; do
+    UUID=$UUID`dd if=/dev/urandom bs=1 count=1 2> /dev/null | hexdump -e '1/1 ":%02x"'`
+  done
+fi
+
+# tunneldigger setup
+uci set tunneldigger.$IFACE=broker
+uci -q delete tunneldigger.$IFACE.address
+uci add_list tunneldigger.$IFACE.address=$TUNNEL_SERV
+uci set tunneldigger.$IFACE.uuid=$UUID
+uci set tunneldigger.$IFACE.interface=$IFACE
+uci set tunneldigger.$IFACE.broker_selection=usage
+uci set tunneldigger.$IFACE.enabled=1
+
+# network setup
+uci set network.${IFACE}_dev=device
+uci set network.${IFACE}_dev.macaddr=$MAC
+uci set network.${IFACE}_dev.name=$IFACE
+
+uci set network.$IFACE=interface
+uci set network.$IFACE.proto=dhcp
+uci set network.$IFACE.ifname=$IFACE
+
+uci set network.olsr_tunnel_$IFACE_ipv4=rule
```

Comment:
The freifunk-policy-routing script will add the ip rules. We don't have to add the rules here.

Reply:
Oh hey, that's cool.  But when is the policy routing script called?  As is it, the router does not need to be restarted when installing the package.  It just works right away.  I would really like to keep this feature.